dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/Clients/GeminiTokenCounterClient.cs

```diff
@@ -1,43 +1,46 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
+using System;
 using System.Net.Http;
 using System.Text.Json.Nodes;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 
-namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI;
+namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI.Core;
 
 /// <summary>
 /// Represents a client for token counting gemini model.
 /// </summary>
 internal sealed class GeminiTokenCounterClient : ClientBase
 {
     private readonly string _modelId;
+    private readonly Uri _tokenCountingEndpoint;
 
     /// <summary>
     /// Represents a client for token counting gemini model.
     /// </summary>
     /// <param name="httpClient">HttpClient instance used to send HTTP requests</param>
     /// <param name="modelId">Id of the model to use to counting tokens</param>
     /// <param name="httpRequestFactory">Request factory for gemini rest api or gemini vertex ai</param>
-    /// <param name="endpointProvider">Endpoints provider for gemini rest api or gemini vertex ai</param>
+    /// <param name="tokenCountingEndpoint">The endpoint for token counting</param>
     /// <param name="logger">Logger instance used for logging (optional)</param>
     public GeminiTokenCounterClient(
         HttpClient httpClient,
         string modelId,
         IHttpRequestFactory httpRequestFactory,
-        IEndpointProvider endpointProvider,
+        Uri tokenCountingEndpoint,
```

Comment:
Leave the token counting endpoint internal, are you adding this parameter just for testing?

Follows the rule that if a code is used only for testing don't add it.

If you really need to override the endpoint use the `HttpStub` (preferred) or an `internal` constructor visible only for the UnitTests.

Reply:
Honestly this client is not used because we don't have appropriate interface. But URI is passed via parameters with same reason like in other clients, we have different endpoint for vertex and google.